File: src/com/rian/difficultycalculator/beatmap/timings/SampleControlPointManager.java
package com.rian.difficultycalculator.beatmap.timings;

import com.rian.difficultycalculator.beatmap.constants.SampleBank;

/**
 * A manager for sample control points.
 */
public class SampleControlPointManager extends ControlPointManager<SampleControlPoint> {
    public SampleControlPointManager() {
        super(new SampleControlPoint(0, SampleBank.normal, 100, 0));
    }

    @Override
    public SampleControlPoint controlPointAt(int time) {
        return binarySearchWithFallback(time);
    }
}

File: src/com/rian/osu/beatmap/timings/SampleControlPointManager.kt
package com.rian.osu.beatmap.timings

import com.rian.osu.beatmap.constants.SampleBank

/**
 * A manager for [SampleControlPoint]s.
 */
class SampleControlPointManager : ControlPointManager<SampleControlPoint>(
    SampleControlPoint(0.0, SampleBank.Normal, 100, 0)
) {
    override fun controlPointAt(time: Double) = binarySearchWithFallback(time, controlPoints.getOrNull(0) ?: defaultControlPoint)
}